### ISSUE TITLE
SFTP file stream not working with System.IO.Compression.Zip

### ISSUE THREAD

**Issue by S-Jindal** [sshnet/SSH.NET#1018]
I am getting the following error when trying to instantiate a [ZipArchive ](https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.ziparchive.-ctor?view=net-6.0#system-io-compression-ziparchive-ctor(system-io-stream-system-io-compression-ziparchivemode)) object using Sftp file stream. 
```
message='Exception: System.IO.InvalidDataException: Central Directory corrupt.
[2022-10-21T14:16:36.733Z]  ---> System.IO.IOException: Zip file corrupt: unexpected end of stream reached.
[2022-10-21T14:16:36.734Z]    at System.IO.Compression.ZipHelper.ReadBytes(Stream stream, Byte[] buffer, Int32 bytesToRead)
[2022-10-21T14:16:36.734Z]    at System.IO.Compression.ZipHelper.SeekBackwardsAndRead(Stream stream, Byte[] buffer, Int32& bufferPointer)
[2022-10-21T14:16:36.736Z]    at System.IO.Compression.ZipHelper.SeekBackwardsToSignature(Stream stream, UInt32 signatureToFind, Int32 maxBytesToRead)
[2022-10-21T14:16:36.736Z]    at System.IO.Compression.ZipArchive.ReadEndOfCentralDirectory()
[2022-10-21T14:16:36.737Z]    --- End of inner exception stack trace ---
[2022-10-21T14:16:36.737Z]    at System.IO.Compression.ZipArchive.ReadEndOfCentralDirectory()
[2022-10-21T14:16:36.738Z]    at System.IO.Compression.ZipArchive..ctor(Stream stream, ZipArchiveMode mode, Boolean leaveOpen, Encoding entryNameEncoding)
[2022-10-21T14:16:36.739Z]    at System.IO.Compression.ZipArchive..ctor(Stream stream, ZipArchiveMode mode, Boolean leaveOpen)
```

Looking at the backtrace, looks like when trying to read from the end of file it is getting this error: **Zip file corrupt: unexpected end of stream reached.**

my code is simple:
```
using(var remoteStream = sftpclient.OpenRead(filepath)
using(var zip = new ZipArchive(stream: remoteStream,  mode: ZipArchiveMode.Read, leaveOpen: true))
{
...
}
```

**Comment by S-Jindal** [comment]
@drieseng  would you be able to help here?  I think you can easily reproduce it on any system with the code above.

**Comment by zybexXL** [comment]
this is fixed in PR #910
